### DISCUSSION TITLE
Checking availability of flow and/or level data before downloading

### DISCUSSION THREAD

**Discussion by rchlumsk** [ropensci/tidyhydat#166] [Q&A] [answered]
This may be a naive question, but is there a way to check whether a given station has flow data or level data prior to calling `hy_daily` or other download function? I know WSC has the 'Data Availability' attribute with either "Flow", "Level", or "Flow and Level", but I don't see that information in the `hy_stations` or `allstations` output (unless I am missing it, which is totally possible).

Thanks! Appreciate any insight on this.

Rob

**Comment by boshek** [comment, marked as answer]
Definitely deleted the wrong one. Not a naive question at all. Great idea. I think the best approach would be to query the data a la:

``` r
library(tidyhydat)
library(dbplyr)
library(dplyr)


hy_check_for_data <- function(station_number, hydat_table) {
  hydat_con <- hy_src()

  tbl(hydat_con, hydat_table) %>%
    filter(STATION_NUMBER %in% station_number) %>%
    group_by(STATION_NUMBER) %>%
    summarise(
      nrows = n(),
      min_year = min(YEAR, na.rm = FALSE),
      max_year = max(YEAR, na.rm = FALSE)
    )
}

hy_check_for_data(c("08MF005", "fffff"), "DLY_FLOWS")
#> # Source:   lazy query [?? x 4]
#> # Database: sqlite 3.36.0
#> #   [C:\Users\salbers\AppData\Local\tidyhydat\tidyhydat\Hydat.sqlite3]
#>   STATION_NUMBER nrows min_year max_year
#>   <chr>          <int>    <int>    <int>
#> 1 08MF005         1294     1912     2019
```
Then inside the function once you have it in R do a `setdiff` on what is returned and figure out someway to communicate that "fffff" isn't in the database (maybe add a row with NA's? With a little work this absolutely could be a useful function inside tidyhydat. The nrows columns is a bit misleading as it returns the rows of the raw table rather than the tidied table.

**Reply by rchlumsk** [reply]
Very neat. When I have some time I will tinker with this a bit and see how it can best be setup, and possibly sent back as a pull request. Thanks for the insight on this

**Reply by boshek** [reply]
The other option would be to write the SQL, embed that in the function and deal with it like that. I'm not certain but I bet you could tune the SQL query a bit more finely than the dplyr code.
